core/App/components/misc/ConnectionNotification.tsx

```diff
@@ -0,0 +1,119 @@
+import { useNavigation } from '@react-navigation/core'
+import { StackNavigationProp } from '@react-navigation/stack'
+import React, { useState } from 'react'
+import { useTranslation } from 'react-i18next'
+import { StyleSheet, Text, View } from 'react-native'
+import Icon from 'react-native-vector-icons/MaterialCommunityIcons'
+
+import { useTheme } from '../../contexts/theme'
+import { RootStackParams, Screens, Stacks } from '../../types/navigators'
+import PopupModal from '../modals/PopupModal'
+
+import { InfoBoxType } from './InfoBox'
+import UnorderedList from './UnorderedList'
+
+interface ConnectionNotificationProps {
+  connectionID?: string
+}
+
+const ConnectionNotification: React.FC<ConnectionNotificationProps> = ({ connectionID }) => {
+  const { t } = useTranslation()
+  const { ListItems, ColorPallet, TextTheme } = useTheme()
+  const [infoCardVisible, setInfoCardVisible] = useState(false)
+
+  const settingsNavigation = useNavigation<StackNavigationProp<RootStackParams>>()
+
+  const styles = StyleSheet.create({
+    modalCenter: {
+      flex: 1,
+      justifyContent: 'center',
+      alignItems: 'center',
+      backgroundColor: 'rgba(0,0,0,0.5)',
+      padding: 20,
+    },
+    notifyTextContainer: {
+      borderLeftColor: ColorPallet.brand.highlight,
+      borderLeftWidth: 10,
+      flex: 1,
+      paddingLeft: 10,
+      paddingVertical: 15,
+      marginVertical: 15,
+    },
+    fakeLink: {
+      ...TextTheme.normal,
+      ...ListItems.recordLink,
+      textDecorationLine: 'underline',
+    },
+    row: {
+      flexDirection: 'row',
+    },
+    notifyTitle: {
+      ...TextTheme.title,
+      marginBottom: 5,
+    },
+    notifyText: {
+      ...TextTheme.normal,
+      marginVertical: 5,
+    },
+    notifyTextList: {
+      marginVertical: 6,
+    },
+    informationIcon: {
+      color: ColorPallet.brand.link,
+      marginLeft: 10,
+    },
+  })
+
+  function toggleInfoCard() {
+    setInfoCardVisible(!infoCardVisible)
+  }
+
+  function goToSettings() {
```

Comment:
While I'd be more specific (navigateToSettings) with the name I have no problem with this.